src/utils/sleep.ts

```diff
@@ -0,0 +1,12 @@
+interface Timeout extends PromiseLike<void> {
+  clear?: () => void;
+}
+
+export const sleep = (ms: number): Timeout => {
```

Comment:
Vitest doesn't support spying on a function which is being tested against within the same file.

eg. we're testing `throwOnTimeout` while expecting `sleep` to be spied on.